fs/zipfs.py

```diff
@@ -444,3 +446,12 @@ def readbytes(self, path):
         zip_name = self._path_to_zip_name(path)
         zip_bytes = self._zip.read(zip_name)
         return zip_bytes
+
+    def geturl(self, path, purpose="download"):
+        # type: (Text, Text) -> Text
+        if purpose == "fs" and isinstance(self._file, six.string_types):
+            quoted_file = url_quote(self._file)
+            quoted_path = url_quote(path)
+            return "zip://{}!/{}".format(quoted_file, quoted_path)
```

Comment:
Given the commonality between tarfs and zipfs in building this "blah://foo!/bar" path, I wonder if it's worth moving part of this method to `_url_tools.py` ? :man_shrugging:

Reply:
Well, marginal gain here. 

Let's do the math: collapse them into one function. (3 + 3 - 2) = 4, meaning 4 less lines. To write a one function into _url_tools.py: 3 lines + 1 function def line + 1 type declaration + 1 docstring + its unit test code lines > 6 lines. In the end, we increase the code base by 2 lines.

Reply:
Fair enough, I was just wondering if it was functionality that might be useful to other FSes too in future. But we can always refactor it later if needed.

Reply:
Indeed!